src/repository.c

```diff
@@ -177,7 +177,7 @@ void git_repository_free(git_repository *repo)
 	git__free(repo->ident_name);
 	git__free(repo->ident_email);
 
-	if (repo->hook_payload != NULL && repo->hook_payload_free != NULL) {
+	if (repo->hook_payload_free != NULL) {
```

Comment:
Okay, so you already do what I wanted in a later commit. Never mind, then :)

Reply:
Yep, will fixup though, it's confusing.